.automation/mirror-labels-to-pr.py

```diff
@@ -0,0 +1,59 @@
+import json
+import re
+import sys
+from github import Github
+
+# Get inputs from shell
+token = sys.argv[1]
+repository = sys.argv[2]
+event_path = sys.argv[3]
+
+# Authenticate with Github using our token
+g = Github(token)
+
+# Initialize repo
+repo = g.get_repo(repository)
+
+# --- Extract PR from event JSON ---
+# `check_run`/`check_suite` does not include any direct reference to the PR
+# that triggered it in its event JSON.
+# We have to extrapolate it using the head SHA that *is* there.
+with open(event_path) as f:
+    event = json.load(f)
+
+# Get head SHA from event JSON
+pr_head_sha = event["check_suite"]["head_sha"]
+
+# Get all repo pull requests
+repo_pull_requests = repo.get_pulls()
+
+# Find the repo PR that matches the head SHA we found
+for potential_pr in repo_pull_requests:
+    if potential_pr.head.sha == pr_head_sha:
+        pr = potential_pr
```

Comment:
Use a dict comprehension for this:

```python
pr = {pr.head.sha: pr for pr in repo_pull_requests}[pr_head_sha]
```

This has the benefit of throwing a clear exception when there is no mathc.